nixos/lib/test-driver/test_driver/logger.py

```diff
@@ -77,9 +79,14 @@ def subtest(self, name: str, attributes: Dict[str, str] = {}) -> Iterator[None]:
         self.tests.setdefault(name, self.TestCaseState())
         self.currentSubtest = name
 
+        tic = time.monotonic()
         yield
+        toc = time.monotonic()
 
+        time_difference = toc - tic
+        self.tests[self.currentSubtest].total_secs = time_difference
         self.currentSubtest = old_test
+        self.tests[self.currentSubtest].nested_secs += time_difference
```

Comment:
Nested subtests will be flatten, but due to https://github.com/kyrus/python-junit-xml/blob/master/junit_xml/__init__.py#L135, the total time is calculated as the sum of all subtests, thus the time of subtests must be calculated without the time of inner subtests.